dashboard/app/public_json_api.go

```diff
@@ -170,3 +177,141 @@ func GetJSONDescrFor(page interface{}) ([]byte, error) {
 	}
 	return json.MarshalIndent(res, "", "\t")
 }
+
+type coveredBlock struct {
+	FromLine int `json:"from_line"`
+	FromCol  int `json:"from_column"`
+	ToLine   int `json:"to_line"`
+	ToCol    int `json:"to_column"`
+}
+
+type functionCoverage struct {
+	FuncName     string          `json:"func_name"`
+	Instrumented int             `json:"total_blocks"`
+	Blocks       []*coveredBlock `json:"covered_blocks"`
+}
+
+type fileCoverage struct {
+	FilePath  string              `json:"file_path"`
+	Functions []*functionCoverage `json:"functions"`
+}
+
+func writeExtAPICoverageFor(ctx context.Context, w io.Writer, ns, repo string) error {
```

Comment:
Nit: let's split this function into the code that actually generates the JSON output and the (very small) wrapper that compresses its output stream. It will be cleaner.

Reply:
#5743 will do it